src/orion/core/worker/experiment.py

```diff
@@ -118,10 +117,13 @@ def __init__(self, name, version=None, mode="r"):
         self.algorithms = None
         self.working_dir = None
         self.producer = {}
-        self._storage = get_storage()
```

Comment:
Is this to avoid pickling issues?

Reply:
yes

Reply:
With more testing, simply making `_storage` as property method can introduce issues:
- For the `workon` function, since it calls `update_singletons` inside it, make `get_storage()` outside the call will have different storage instance and leading the failure of some method calls (related to db of course) of the returned `ExperimentClient`.
- Similarly, although the `ExperimentClient.workon` does not `update_singletons` , but orion api provides such options, if user build `ExperimentClient` with one storage configure, call its `workon` method, and then `update_singletons`, will make user fail if they simply call `ExperimentClient` db query related apis.

I made some change as below, it may be able to solve above 2 issues, but will introduce a new issue, if user call `ExperimentClient.workon` with `n_workers > 2` first time, it will be okay, but if after this first call, user call `ExperimentClient.workon` db related apis, such as `fetch_trials`, and then call `ExperimentClient.workon` with `n_workers > 2` again, there will be un-picklable issue since ExperimentClient will now have a storage instance which may not un-picklable, like mongodb.
```
        self._istorage = None

    @property
    def _storage(self):
        if not self._istorage:
            self._istorage = get_storage()
        return self._istorage
```

To solve this, I am now thinking to **clear** the storage instance property before `joblib parallel` and **reset** it back after that inside `ExperimentClient.workon`. But it is a little bit tricky, especially when there is error between the **clear** and **reset**. What do you think?

Reply:
Could we implement instead a method to pickled the experiment which ignores the storage? We could implement [__getstate__](https://docs.python.org/3/library/pickle.html#object.__getstate__) for instance.

Reply:
This can be an option. Let me give it a try.